Lib/dataclasses.py

```diff
@@ -1549,11 +1550,22 @@ class C(Base):
         seen.add(name)
         annotations[name] = tp
 
+    def annotate_method(format):
+        from typing import Any, _convert_to_source
+        ann_dict = {
+            ann: Any if t is any_marker else t
+            for ann, t in annotations.items()
+        }
+        if format == 1 or format == 2:
+            return ann_dict
+        else:
+            return _convert_to_source(ann_dict)
+
```

Comment:
Conceptually I was thinking something along these lines:

```python
    def annotate_method(format):
        fmt = annotationlib.Format(format)
        typing = sys.modules.get("typing")

        if fmt == annotationlib.Format.FORWARDREF and typing is None:
            # Not sure what other arguments to ForwardRef would be correct here
            typing_any = annotationlib.ForwardRef("typing.Any", ...)
            ann_dict = {
                ann: typing_any
                if t is any_marker 
                else t
                for ann, t in annotations.items()
            }
        else:
            from typing import Any, _convert_to_source
            ann_dict = {
                ann: Any if t is any_marker else t
                for ann, t in annotations.items()
            }
            if fmt == annotationlib.Format.SOURCE:
                ann_dict = _convert_to_source(ann_dict)
            
        return ann_dict
```

This prevents the import of `typing` when the dataclass is created but it does have the side effect of putting `ForwardRef("typing.Any", ...)` as the type on the dataclass fields in this case, previously this would have been the string "typing.Any".

Ideally you'd want this ForwardRef to `.evaluate` to `typing.Any` if that was called but I'm not sure there's a *simple* way to guarantee it would be the `typing.Any` object?

Reply:
Oh, somewhat unhelpfully I failed to capture the whole block here - it was this block:

https://github.com/python/cpython/blob/642f8498845592bf8e7197f1aedfd0c894bbe589/Lib/dataclasses.py#L1553-L1562

Reply:
Good idea, will try this out. Thank you!